### DISCUSSION TITLE
Implement JSON and search

### DISCUSSION THREAD

**Discussion by MagnusRef** [valkey-io/valkey#212] [General]
Would love to see that Valkey implemented stack components like storing JSON as well as indexing and Searching of JSON documents.

This discussion is related to the other discussions that asks for redis-stack features:  [Graph](https://github.com/orgs/valkey-io/discussions/120) and [Time-series ](https://github.com/orgs/valkey-io/discussions/124)

**Comment by rlunar** [comment]
+1 on this request, I'd just recommend to break it apart into 2 different requests

**Comment by zuiderkwast** [comment]
I believe we can consider it if anyone is willing to contribute it. Compatible with the redis stack ones, I assume you want?

Apparently, they can be implemented as modules, so they don't even have to be approved by us. :grin:

Modules are good either way, for separation.

**Comment by yangbodong22011** [comment]
If we want to contribute RedisJSON-like capabilities to valkey, should it be in the form of a Module? Where should its repository be placed?

**Reply by zuiderkwast** [reply]
I think we can put it under the valkey organization, or even include it in the main repo, but it can still be a module to keep a good separation. WDYT?

**Reply by yangbodong22011** [reply]
We'd love to donate these two modules to valkey which are ready in Alibaba cloud and i agree that the code is organized into separate module repos, here we need two repos at valkey organization (names to be determined):
- ValkeyJSON: similar to RedisJSON
- ValkeyHash: supports hash field expiration

**Reply by mkmkme** [reply]
Hey there!
I'm interested in contributing to valkey. Do I understand correctly that RedisJSON you're talking about is this one? https://github.com/RedisJSON/RedisJSON

If so, I'd love to fork it and replace `redis` to `valkey` so far if nobody got it yet. WDYT?

UPDATE: IIUC it was forked already - https://github.com/orgs/valkey-io/discussions/108#discussioncomment-9117139

**Reply by zuiderkwast** [reply]
@mkmkme No, RedisJSON is not open source. If we fork it, the fork will need to have the same non-open-source license. We don't want that.

There is one old version of RedisJSON which was forked before the license was changed to non-open source some years ago. Amazon has another implementation. Alibaba has a third implementation. They are all supposed to be API compatible with RedisJSON IIUC, but we don't know the details about the Amazon and Alibaba versions because I'm not sure they're available yet(?).

**Reply by martinrvisser** [reply]
Following up on this, are there any plans to add one of these to upstream Valkey to be maintained or made otherwise available? And modules in general?
